File: src/src/org/github/olloginov/ideataskbrowser/actions/RefreshListAction.java
package org.github.olloginov.ideataskbrowser.actions;

import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.components.ServiceManager;
import com.intellij.openapi.project.Project;
import org.github.olloginov.ideataskbrowser.TaskBrowser;

public class RefreshListAction extends AnActionImpl {
    public RefreshListAction() {
        super("RefreshList");
    }

    @Override
    protected boolean isEnabled(Project project) {
        return true;
    }

    @Override
    public void actionPerformed(AnActionEvent e) {
        Project project = e.getProject();
        if (project == null) {
            return;
        }

        TaskBrowser browser = ServiceManager.getService(e.getProject(), TaskBrowser.class);
        if (browser != null) {
            browser.refresh();
        }
    }
}

File: src/main/kotlin/org/github/olloginov/ideataskbrowser/actions/RefreshListAction.kt
package org.github.olloginov.ideataskbrowser.actions

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.project.Project
import org.github.olloginov.ideataskbrowser.TaskBrowser

class RefreshListAction(
    noop: Boolean
) : AnActionImpl("RefreshList", noop) {
    override fun isEnabled(project: Project): Boolean {
        return true
    }

    override fun actionPerformedNow(e: AnActionEvent) {
        e.project?.getService(TaskBrowser::class.java)?.refresh()
    }
}
